sample_projects/bin_required_features/src/bin/some_binary_with_rand.rs

```diff
@@ -0,0 +1,7 @@
+use rand::prelude::*;
+
+pub fn main() {
+    let mut rng = rand::thread_rng();
+    let y: f64 = rng.gen();
+    println!("The binary that should NOT be compiled found the random number {}", y);
```

Comment:
Let's instead use https://doc.rust-lang.org/std/macro.compile_error.html to make building this package and error so the test is more stringent?

I am happy to merge the Nixpkgs PR in a second per the comment I left, so don't worry that doing the above will make CI fail today.